lib/compiler/passes/generate-js.js

```diff
@@ -499,7 +499,7 @@ function generateJS(ast, options) {
         ip += bodyLength;
 
         if (bodySp !== baseSp) {
-          throw new Error("Body of a loop can't move the stack pointer.");
+          throw new Error("Body of a loop can't move the stack pointer (base=" + baseSp + ", body=" + bodySp + ").");
```

Comment:
I’d format the message like this for more readability: “Body of a loop can't move the stack pointer (before: 5, after: 6).”.

Also please split the concatenation over multiple lines.